js/hotgraphicView.js

```diff
@@ -3,145 +3,145 @@ define([
   'core/js/views/componentView',
   './hotgraphicPopupView'
 ], function(Adapt, ComponentView, HotgraphicPopupView) {
-  'use strict';
 
-  var HotGraphicView = ComponentView.extend({
+  class HotGraphicView extends ComponentView {
 
-    events: {
-      'click .js-hotgraphic-item-click': 'onPinClicked'
-    },
+    events () {
+      return {
+        'click .js-hotgraphic-item-click': 'onPinClicked'
+      }
+    }
+
+    initialize(...args) {
+      super.initialize(...args);
 
-    initialize: function() {
-      ComponentView.prototype.initialize.call(this);
       this.setUpViewData();
       this.setUpModelData();
       this.setUpEventListeners();
       this.checkIfResetOnRevisit();
-    },
+    }
 
-    setUpViewData: function() {
+    setUpViewData() {
       this.popupView = null;
       this._isPopupOpen = false;
-    },
+    }
 
-    setUpModelData: function() {
+    setUpModelData() {
       if (this.model.get('_canCycleThroughPagination') === undefined) {
         this.model.set('_canCycleThroughPagination', false);
       }
-    },
+    }
 
-    setUpEventListeners: function() {
+    setUpEventListeners() {
       this.listenTo(Adapt, 'device:changed', this.reRender);
 
       this.listenTo(this.model.get('_children'), {
         'change:_isActive': this.onItemsActiveChange,
         'change:_isVisited': this.onItemsVisitedChange
       });
-    },
+    }
 
-    reRender: function() {
-      if (Adapt.device.screenSize !== 'large') {
-        this.replaceWithNarrative();
-      }
-    },
-
-    replaceWithNarrative: function() {
-      var NarrativeView = Adapt.getViewClass('narrative');
-
-      var model = this.prepareNarrativeModel();
-      var newNarrative = new NarrativeView({ model: model });
-      var $container = $(".component__container", $("." + this.model.get("_parentId")));
-
-      newNarrative.reRender();
```

Comment:
Were the reRender() and setupNarrative() calls not needed?

Reply:
it seems to work perfectly well without them..! but please do give it a try and make sure I haven't missed anything

Reply:
the deferred `Adapt.trigger('device:resize');` event will cause the narrative component to call `reRender` anyway... and `setupNarrative` is already being triggered by the narrative `postRender` method...